tests/std/tests/P2164R9_views_enumerate/test.cpp

```diff
@@ -0,0 +1,645 @@
+            [[maybe_unused]] const iterator_t<const R> const_defaulted;
+        }
+
+        same_as<iterator_t<R>> auto i        = r.begin();
+        same_as<iterator_t<const R>> auto ci = as_const(r).begin();
+        assert(ci.index() == 0);
+
+        { // Check comparisons
+            assert(ci == ci);
+            assert(!(ci < ci));
+            assert(ci <= ci);
+            assert(!(ci > ci));
+            assert(ci >= ci);
+            assert(ci <=> ci == strong_ordering::equal);
+
+            if constexpr (forward_range<R>) {
+                auto ci2 = ranges::next(ci, 1);
+                assert(ci != ci2);
+                assert(ci < ci2);
+                assert(ci <= ci2);
+                assert(ci2 > ci);
+                assert(ci2 >= ci);
+                assert(ci <=> ci2 == strong_ordering::less);
+                assert(ci2 <=> ci == strong_ordering::greater);
+            }
+
+            if constexpr (constructible_from_nonconst) { // Check cross comparisons
+                assert(ci == i);
+                assert(!(ci < i));
+                assert(ci <= i);
+                assert(!(ci > i));
+                assert(ci >= i);
+                assert(ci <=> i == strong_ordering::equal);
+
+                if constexpr (forward_range<const R>) {
+                    auto i2 = ranges::next(i, 1);
+                    assert(ci != i2);
+                    assert(ci < i2);
+                    assert(ci <= i2);
+                    assert(i2 > ci);
+                    assert(i2 >= ci);
+                    assert(ci <=> i2 == strong_ordering::less);
+                    assert(i2 <=> ci == strong_ordering::greater);
+                }
+            }
+        }
+
+        if constexpr (forward_range<const R>) {
+            assert(*ci++ == expected[0]);
+        } else {
+            STATIC_ASSERT(is_void_v<decltype(i++)>);
+        }
+        assert(*++ci == expected[2]);
+        assert(ci.index() == 2);
+
+        if constexpr (bidirectional_range<const R>) {
+            assert(*ci-- == expected[2]);
+            assert(*--ci == expected[0]);
+        }
+
+        if constexpr (random_access_range<const R>) {
+            ci += 2;
+            assert(*ci == expected[2]);
+
+            ci -= 2;
+            assert(*ci == expected[0]);
+
+            assert(ci[2] == expected[2]);
+
+            const auto ci2 = ci + 2;
+            assert(*ci2 == expected[2]);
+
+            const auto ci3 = 2 + ci;
+            assert(*ci3 == expected[2]);
+
+            const auto ci4 = ci3 - 2;
+            assert(*ci4 == expected[0]);
+
+            const same_as<ranges::range_difference_t<V>> auto diff1 = ci2 - ci;
+            assert(diff1 == 2);
+
+            const same_as<ranges::range_difference_t<V>> auto diff2 = ci - ci2;
+            assert(diff2 == -2);
+
+            if constexpr (sized_sentinel_for<sentinel_t<const V>, iterator_t<const V>>) {
+                const auto i2   = as_const(r).begin();

```

Comment:
After making `random_access_range` and `sized_sentinel_for` non-nested, it would seem better to name this `ci2` because it's from `as_const(r)`. (With the nesting, attempting to do this would shadow line 493.)